crud/common/utils.lua

```diff
@@ -748,4 +751,75 @@ function utils.list_slice(list, start_index, end_index)
     return slice
 end
 
+--- Polls replicas for storage state
+--
+-- @function storage_info
+--
+-- @tparam ?number opts.timeout
+--  Function call timeout
+--
+-- @return a table of storage states by replica uuid.
+function utils.storage_info(opts)
+    local replicasets, err = vshard.router.routeall()
+    if replicasets == nil then
+        return nil, StorageInfoError:new("Failed to get all replicasets: %s", err.err)
+    end
+
+    opts = opts or {}
+
+    local futures_by_replicas = {}
+    local replica_state_by_uuid = {}
+    local async_opts = {is_async = true}
+    local timeout = opts.timeout or const.DEFAULT_VSHARD_CALL_TIMEOUT
+
+    for _, replicaset in pairs(replicasets) do
+        for replica_uuid, replica in pairs(replicaset.replicas) do
+            replica_state_by_uuid[replica_uuid] = {status = "error",
+                                                   is_master = replicaset.master == replica}
+            local ok, res = pcall(replica.conn.call, replica.conn, "_crud.storage_info_on_storage",
+                                  {}, async_opts)
+            if ok then
+                futures_by_replicas[replica_uuid] = res
+            elseif res ~= nil then
+                log.error("Error getting storage info for %s: %s", replica_uuid, res)
+                replica_state_by_uuid[replica_uuid].message = tostring(res)
+            end
```

Comment:
Shouldn't we fill `message` if an error occurs here (if `ok` is `false` disregarding whether `res` is `nil`)?

Reply:
We need to log it, but I'm not sure about usefulness of common error message.
Done.

Reply:
Is it ever possible that we'll get `false, nil` result from `pcall(replica.conn.call, <...>)`? Anyway, even if so, I think that it would be misleading to see `status = 'error'` without any message.

Thanks for the update.